src/pages/ProtectedRoute.tsx

```diff
@@ -0,0 +1,35 @@
+import { ReactNode } from 'react';
+import { Navigate } from 'react-router-dom';
+import { useEffect, useState } from 'react';
+import { getAuth, onAuthStateChanged } from 'firebase/auth';
+import { getAuthToken } from '@/utils/cookieUtils.ts';
+
+const ProtectedRoute = ({ children }: { children: ReactNode }) => {
+  const [isAuthenticated, setIsAuthenticated] = useState<boolean | null>(null);
```

Comment:
Běžne by jsem držel state v nějakém slice, aby se dalo využít i jinde. Ale tím, že appka je uzavřená pro anonymous users tak je to takhle ok.